diracx-db/src/diracx/db/sql/utils/types.py

```diff
@@ -41,3 +39,37 @@ def process_result_value(self, value, dialect) -> bool:
             return False
         else:
             raise NotImplementedError(f"Unknown {value=}")
+
+
+class StringParsingDateTime(types.TypeDecorator):
+    """A DateTime type that also accepts ISO8601 strings.
+
+    IMPORTANT: Naïve timestamps have tzinfo forced to UTC by default! (change with tz=...)
+
+    """
+
+    impl = DateTime()
+    cache_ok = True
+
+    def __init__(self, tz=timezone.utc):
+        self._tz = tz
+
+    def process_bind_param(self, value, dialect):
+        if value is None:
+            return None
+
+        if isinstance(value, str):
+            try:
+                value: datetime = datetime.fromisoformat(value)
+            except ValueError as err:
+                raise ValueError(f"Unable to parse datetime string: {value}") from err
+
+        if not isinstance(value, datetime):
+            raise ValueError(f"Expected datetime or ISO8601 string, but got {value!r}")
+
+        return value
+
+    def process_result_value(self, value, dialect):
+        if isinstance(value, datetime) and not value.tzinfo:
+            value = value.replace(tzinfo=self._tz)
```

Comment:
This is a potentially a landmine that needs to be checked

Reply:
OK I've refactored the class a bit to do the time zone calculations with respect to whether the DB deals in local time or not